### PR TITLE
Frontegg OIDC and SAML Integration Guide

### PR DESCRIPTION
**Pull request by faonbr**

### CODE REVIEW

File: docs/integrations/frontegg/img/frontegg-2.png
Comment (russorat) [inline docs/integrations/frontegg/img/frontegg-2.png:1]: i don't think this image should have the `--region us` in it. i'm guessing this is an old file?

File: docs/integrations/frontegg/img/frontegg-6.png
Comment (russorat) [inline docs/integrations/frontegg/img/frontegg-6.png:1]: this says JumpCloud, is that right?

File: docs/integrations/frontegg/img/launch_ngrok_tunnel.png
Comment (russorat) [inline docs/integrations/frontegg/img/launch_ngrok_tunnel.png:1]: this screenshot should use the latest agent and have an updated domain using either ngrok-free.app/dev or ngrok.app/dev

File: docs/integrations/frontegg/img/ngrok_introspection_frontegg_hooks.png
Comment (russorat) [inline docs/integrations/frontegg/img/ngrok_introspection_frontegg_hooks.png:1]: this says okta, not frontegg

File: docs/integrations/frontegg/sso-oidc.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+### **Option 1**: ngrok CLI
+
+> **Note:** For this tutorial, we assume you have an app running locally (i.e., on localhost:3000) with the ngrok client installed.
+
+1. Launch a terminal
+
+1. Enter the following command to launch an ngrok tunnel with Frontegg SSO:
+
+   ```bash
+   ngrok http 3000 --oidc=Frontegg_OAUTH_URL \
```
Comment (russorat) [inline docs/integrations/frontegg/sso-oidc.md:92]: the standard for env variables IS_ALL_CAPS. can we change this to FRONTEGG_OAUTH_URL? same with the client id and secret.

File: docs/integrations/frontegg/img/frontegg-2.png
Comment (faonbr) [reply docs/integrations/frontegg/img/frontegg-2.png:1]: I didn't updated some of the images yet. Will do to the next update.

File: docs/integrations/frontegg/img/frontegg-6.png
Comment (faonbr) [reply docs/integrations/frontegg/img/frontegg-6.png:1]: This image is supposed to be used in the SAML guide. Gonna remove till I write it.

File: docs/integrations/frontegg/img/launch_ngrok_tunnel.png
Comment (faonbr) [reply docs/integrations/frontegg/img/launch_ngrok_tunnel.png:1]: This image is not used by the OIDC guide. Gonna remove it.

File: docs/integrations/frontegg/img/ngrok_introspection_frontegg_hooks.png
Comment (faonbr) [reply docs/integrations/frontegg/img/ngrok_introspection_frontegg_hooks.png:1]: This is not used by the OIDC guide. gonna remove it.

File: docs/integrations/frontegg/sso-oidc.md (hunk 1 above)
Comment (faonbr) [reply docs/integrations/frontegg/sso-oidc.md:92]: You're right. Fixed.

File: docs/integrations/frontegg/index.mdx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+name: frontegg
+title: Frontegg Integration Hub
+sidebar_label: Frontegg
+description: |
+  Secure access to ngrok tunnels with Frontegg SSO.
```
Comment (russorat) [inline docs/integrations/frontegg/index.mdx:6]:
```suggestion
  Secure access to ngrok endpoints with Frontegg SSO. 
```

File: docs/integrations/frontegg/index.mdx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+name: frontegg
+title: Frontegg Integration Hub
+sidebar_label: Frontegg
+description: |
+  Secure access to ngrok tunnels with Frontegg SSO. 
+  No VPNs, Firewalls, or complex network rules required.
+excerpt: |
+  Secure access to your ngrok tunnels with Frontegg SSO.
```
Comment (russorat) [inline docs/integrations/frontegg/index.mdx:9]:
```suggestion
  Secure access to your ngrok endpoints with Frontegg SSO.
```

File: docs/integrations/frontegg/sso-oidc.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
+---
+description: Use Frontegg OIDC to secure access to ngrok tunnels
```
Comment (russorat) [inline docs/integrations/frontegg/sso-oidc.md:2]:
```suggestion
description: Use Frontegg OIDC to secure access to ngrok endpoints
```

File: docs/integrations/frontegg/sso-oidc.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+---
+
+# Frontegg SSO (OpenID Connect)
+
+---
+
+:::tip TL;DR
+
+To secure access to ngrok with Frontegg Single Sign-On using OpenID Connect:
```
Comment (russorat) [inline docs/integrations/frontegg/sso-oidc.md:11]:
```suggestion
To secure access to ngrok with Frontegg Single Sign-On (SSO) using OpenID Connect:
```

File: docs/integrations/frontegg/sso-oidc.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+1. [Configure Frontegg SSO](#configure-frontegg)
+1. [Configure ngrok](#configure-ngrok)
+1. [Test access to ngrok with Frontegg SSO](#test-sso)
+
+:::
+
+This article details how to configure Frontegg as the primary Identity Provider for ngrok tunnels.
```
Comment (russorat) [inline docs/integrations/frontegg/sso-oidc.md:19]:
```suggestion
This article details how to configure Frontegg as the primary Identity Provider (IDP) for ngrok endpoints.
```

File: docs/integrations/frontegg/sso-oidc.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+By integrating Frontegg SSO with ngrok, you can:
+
+- **Restrict access to ngrok tunnels** only to users authenticated via Frontegg
```
Comment (russorat) [inline docs/integrations/frontegg/sso-oidc.md:22]:
```suggestion
- **Restrict access to ngrok endpoints** only to users authenticated via Frontegg
```

File: docs/integrations/frontegg/sso-oidc.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+- **Use Frontegg security policies and MFA authenticators**.
+- **Use Frontegg's Dashboard to facilitate access to ngrok apps**.
+
+## Supported Features
+
+The ngrok integration with Frontegg supports:
+
+- **SP-Initiated SSO**: In this mode, users access ngrok edges and tunnels and are redirected to Frontegg for authentication.
```
Comment (russorat) [inline docs/integrations/frontegg/sso-oidc.md:30]:
```suggestion
- **SP-Initiated SSO**: In this mode, users access ngrok endpoints and are redirected to Frontegg for authentication.
```

File: docs/integrations/frontegg/sso-oidc.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+
+## Requirements
+
+To configure ngrok tunnels with Frontegg, you must have:
```
Comment (russorat) [inline docs/integrations/frontegg/sso-oidc.md:34]:
```suggestion
To configure an ngrok edge with Frontegg, you must have:
```

File: docs/integrations/frontegg/sso-oidc.md [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+- a Frontegg account with administrative rights to create apps
```
Comment (russorat) [inline docs/integrations/frontegg/sso-oidc.md:36]:
```suggestion
- A Frontegg account with administrative rights to create apps
```

File: docs/integrations/frontegg/sso-oidc.md [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+- an [ngrok Enterprise Account](https://ngrok.com/pricing) with an authtoken or admin access to configure edges with OpenID Connect.
```
Comment (russorat) [inline docs/integrations/frontegg/sso-oidc.md:37]:
```suggestion
- An [ngrok Enterprise Account](https://ngrok.com/pricing) with an authtoken or admin access to configure edges with OpenID Connect.
```

File: docs/integrations/frontegg/sso-saml.md [hunk 12]
```diff
@@ -0,0 +1,201 @@
+---
+description: Use Frontegg SAML to secure access to ngrok tunnels
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:2]:
```suggestion
description: Use Frontegg SAML to secure access to ngrok endpoints
```

File: docs/integrations/frontegg/sso-saml.md [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+
+1. [Download the IDP metadata](#idp-metadata)
+1. [Configure ngrok](#configure-ngrok)
+1. [Configure Frontegg](#configure-frontegg)
+1. [Update Frontegg Login Method](#frontegg-login)
+1. [Test access to ngrok with Frontegg SSO](#test-sso)
+
+:::
+
+This article details how to configure Frontegg as the primary Identity Provider for ngrok tunnels.
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:21]:
```suggestion
This article details how to configure Frontegg as the primary Identity Provider for ngrok endpoints.
```

File: docs/integrations/frontegg/sso-saml.md [hunk 14]
```diff
@@ -0,0 +1,201 @@
...
+By integrating Frontegg SSO with ngrok, you can:
+
+- **Restrict access to ngrok tunnels** only to users authenticated via Frontegg
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:24]:
```suggestion
- **Restrict access to ngrok endpoints** only to users authenticated via Frontegg
```

File: docs/integrations/frontegg/sso-saml.md [hunk 15]
```diff
@@ -0,0 +1,201 @@
...
+- **Use Frontegg security policies and MFA authenticators**.
+- **Use Frontegg's Dashboard to facilitate access to ngrok apps**.
+
+## Requirements
+
+To configure ngrok tunnels with Frontegg, you must have:
+
+- a Frontegg account with administrative rights to create apps
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:32]:
```suggestion
- A Frontegg account with administrative rights to create apps
```

File: docs/integrations/frontegg/sso-saml.md [hunk 16]
```diff
@@ -0,0 +1,201 @@
...
+- an [ngrok Enterprise Account](https://ngrok.com/pricing) with an authtoken or admin access to configure edges with SAML.
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:33]:
```suggestion
- An [ngrok Enterprise Account](https://ngrok.com/pricing) with an authtoken or admin access to configure edges with SAML.
```

File: docs/integrations/frontegg/sso-saml.md [hunk 17]
```diff
@@ -0,0 +1,201 @@
...
+    - CLIENT_ID: The client ID value you copied before.
+    - API_KEY: The API key value you copied before.
+
+1. Copy the value of the **token** attribute from the response.
+
+1. In the same terminal window, run the following command to download the metadata XML file from Frontegg:
+
+   ```bash
+   curl --location --request POST 'https://api.frontegg.com/oauth/resources/configurations/saml/v1/https%3A%2F%2Ftemporary' \
+   --header 'frontegg-vendor-host: YOUR-FRONTEGG-HOST-URL' \
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:71]:
```suggestion
   --header 'frontegg-vendor-host: YOUR_FRONTEGG_HOST_URL' \
```

File: docs/integrations/frontegg/sso-saml.md [hunk 18]
```diff
@@ -0,0 +1,201 @@
...
+   --header 'Content-Type: application/json' \
+   --data-raw '{
+      "acsUrl": "https://temporary",
+      "entityId": "https://temporary"
+   }'
+   ```
+
+   **Note**: Replace the following with values copied on previous steps:
+
+   - YOUR-FRONTEGG-HOST-URL: The value of the **Domain name** from the **Env settings** > **Domains** tab.
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:82]:
```suggestion
   - YOUR_FRONTEGG_HOST_URL: The value of the **Domain name** from the **Env settings** > **Domains** tab.
```

File: docs/integrations/frontegg/sso-saml.md [hunk 19]
```diff
@@ -0,0 +1,201 @@
...
+1. Copy both the **Entity ID** and the **ACS URL** values for later use.
+
+
+### **Step 3**: Configure Frontegg {#configure-frontegg}
+
+1. In the same terminal window in which you ran the previous commands, run the following command to create a SAML configuration related to your ngrok SAML edge:
+
+   ```bash
+   curl --location --request POST 'https://api.frontegg.com/oauth/resources/configurations/saml/v1/URL-ENCODED-NGROK-ENTITY-ID' \
+   --header 'frontegg-vendor-host: YOUR-FRONTEGG-HOST-URL' \
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:119]:
```suggestion
   --header 'frontegg-vendor-host: YOUR_FRONTEGG_HOST_URL' \
```

File: docs/integrations/frontegg/sso-saml.md [hunk 20]
```diff
@@ -0,0 +1,201 @@
...
+   --header 'Authorization: Bearer TOKEN' \
+   --header 'Content-Type: application/json' \
+   --data-raw '{
+      "acsUrl": "NGROK-ACS-URL",
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:123]:
```suggestion
      "acsUrl": "NGROK_ACS_URL",
```

File: docs/integrations/frontegg/sso-saml.md [hunk 21]
```diff
@@ -0,0 +1,201 @@
...
+      "entityId": "NGROK-ENTITY-ID-URL"
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:124]:
```suggestion
      "entityId": "NGROK_ENTITY_ID_URL"
```

File: docs/integrations/frontegg/sso-saml.md [hunk 22]
```diff
@@ -0,0 +1,201 @@
...
+   }'
+   ```
+
+   **Note**: Replace the following with the values copied on previous steps:
+
+   - URL-ENCODED-NGROK-ENTITY-ID: URL-Enconded value of the ngrok **Entity ID*** copied from the edge SAML configuration.
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:130]:
```suggestion
   - URL_ENCODED_NGROK_ENTITY_ID: URL-Encoded value of the ngrok **Entity ID*** copied from the edge SAML configuration.
```

File: docs/integrations/frontegg/sso-saml.md [hunk 23]
```diff
@@ -0,0 +1,201 @@
...
+   - YOUR-FRONTEGG-HOST-URL: The value of the **Domain name** from the **Env settings** > **Domains** tab.
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:131]:
```suggestion
   - YOUR_FRONTEGG_HOST_URL: The value of the **Domain name** from the **Env settings** > **Domains** tab.
```

File: docs/integrations/frontegg/sso-saml.md [hunk 24]
```diff
@@ -0,0 +1,201 @@
...
+   - TOKEN: The Frontegg token you copied before.
+   - NGROK-ACS-URL: The value of the ngrok **Entity ID** copied from the edge SAML configuration.
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:133]:
```suggestion
   - NGROK_ACS_URL: The value of the ngrok **Entity ID** copied from the edge SAML configuration.
```

File: docs/integrations/frontegg/sso-saml.md [hunk 25]
```diff
@@ -0,0 +1,201 @@
...
+   - NGROK-ENTITY-ID-URL: The value of the ngrok **ACS URL** copied from the edge SAML configuration.
```
Comment (russorat) [inline docs/integrations/frontegg/sso-saml.md:134]:
```suggestion
   - NGROK_ENTITY_ID_URL: The value of the ngrok **ACS URL** copied from the edge SAML configuration.
```

**Pull request merged**